### PR TITLE
fix: wash dev no duplicate components, ignore generated files

### PR DESCRIPTION
**Pull request by brooksmtownsend**
## Feature or Problem
This PR fixes two issues in `wash dev`
1. Previously, when generating a dependency for something like `wasi:keyvalue/store` and `wasi:keyvalue/atomics`, we would generate two capability components in the manifest. Now, I use a hashmap to remove the duplicate component during manifest creation.
    1. Additionally, these links are now combined into one link
1. We're watching paths recursively, so generated `target` / `node_modules` type directories triggers many many rebuilds. I added an `--ignore-dir` flag so that users can submit their own, but I made sure to ignore target + build to start.

### CODE REVIEW

File: crates/wash-cli/src/dev.rs [hunk 1]
```diff
@@ -1514,7 +1544,7 @@ async fn run_dev_loop(
     let wit_implied_deps = discover_dependencies_from_wit(resolve, world_id)
         .context("failed to resolve dependent components")?;
     eprintln!(
-        "Detected component dependencies: {:?}",
+        "🔎 Detected component dependencies: {:?}",
```
Comment (lachieh) [inline crates/wash-cli/src/dev.rs:1547]: `emoji::SEARCH`?

File: crates/wash-cli/src/dev.rs [hunk 2]
```diff
@@ -1875,6 +1923,10 @@ pub async fn handle_command(
 
     // Watch FS for changes and listen for Ctrl + C in tandem
     eprintln!("👀 watching for file changes (press Ctrl+c to stop)...");
+    let _ = reload_rx.try_recv();
+    let _ = reload_rx.try_recv();
+    let _ = reload_rx.try_recv();
+    let _ = reload_rx.try_recv();
```
Comment (lachieh) [inline crates/wash-cli/src/dev.rs:1929]: What does this do?

Comment (brooksmtownsend) [reply crates/wash-cli/src/dev.rs:1929]: Ugh this was a leftover. I threw in a bunch to try and make sure we don't immediately rebuild but I think we need at most one

File: crates/wash-cli/src/dev.rs (hunk 1 above)
Comment (brooksmtownsend) [reply crates/wash-cli/src/dev.rs:1555]: I actually fixed this but I must not have force pushed.

File: crates/wash-cli/src/dev.rs (hunk 2 above)
Comment (brooksmtownsend) [reply crates/wash-cli/src/dev.rs:1937]: I actually fixed this but I must not have force pushed.

Comment (lachieh) [reply crates/wash-cli/src/dev.rs:1937]: i don't believe you

File: crates/wash-cli/src/dev.rs [hunk 3]
```diff
@@ -91,6 +92,11 @@ pub struct DevCommand {
     #[clap(name = "code-dir", long = "work-dir", env = "WASH_DEV_CODE_DIR")]
     pub code_dir: Option<PathBuf>,
 
+    /// Directories to ignore when watching for changes. This should be set
+    /// to directories where generated files are placed, such as `target/` or `dist/`
```
Comment (vados-cosmonic) [inline crates/wash-cli/src/dev.rs:96]: Should the first sentence be separated from the second? And maybe an indication it can be specified multiple times

**Review by vados-cosmonic — APPROVED** [review]
Approved pending the changes that were already pointed out! My thing is just a nit.

File: crates/wash-cli/src/dev.rs (hunk 3 above)
Comment (brooksmtownsend) [reply crates/wash-cli/src/dev.rs:96]: Good idea, I'll incorporate this in #3140 just to keep merging simple!

**Pull request merged**